managers/tests/test_views.py

```diff
@@ -258,119 +258,85 @@ def test_delete_project_function_view_should_not_delete_project_on_non_admin_req
         self.assertEqual(Project.objects.all().count(), 1)
 
 
-class ManageTaskActivitiesInProjectViewTests(TestCase):
+class ManageTaskActivitiesTests(TestCase):
     def setUp(self):
         super().setUp()
         self.user_admin = AdminUserFactory()
         self.client.force_login(self.user_admin)
-        self.default_task_activity_1 = TaskActivityTypeFactory(is_default=True)
-        self.default_task_activity_2 = TaskActivityTypeFactory(is_default=True)
-        self.not_default_task_activity = TaskActivityTypeFactory()
-
         self.project = ProjectFactory()
-        self.url = reverse("project-task-activities", kwargs={"pk": self.project.pk})
+        self.task_activities = TaskActivityTypeFactory()
+        self.url = reverse("project-task-activities")
 
-    def test_manage_task_activities_in_project_view_should_raise_404_for_manager_not_of_current_project(self):
-        other_manager = ManagerUserFactory()
-        self.client.force_login(other_manager)
+    def test_manage_task_activities_view_should_return_200_status_code_when_user_is_manager_or_admin(self):
+        manager = ManagerUserFactory()
+        admin = AdminUserFactory()
+        self.client.force_login(manager)
 
         response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
 
-        self.assertEqual(response.status_code, 404)
-
-    def test_manage_task_activities_in_project_view_should_get_200_when_user_is_manager_of_current_project(self):
-        project_manager = ManagerUserFactory()
-        self.project.managers.add(project_manager)
-        self.client.force_login(project_manager)
-
+        self.client.force_login(admin)
         response = self.client.get(self.url)
-
         self.assertEqual(response.status_code, 200)
 
-    def test_manage_task_activities_in_project_view_should_redirect_on_employee_get(self):
+    def test_manage_task_activities_should_redirect_on_employee_get(self):
         user_employee = UserFactory()
-        self.project.members.add(user_employee)
-        self.client.force_login(user_employee)
 
+        self.client.force_login(user_employee)
         response = self.client.get(self.url)
 
         self.assertEqual(response.status_code, 302)
 
-    def test_manage_task_activities_view_should_show_all_task_activities_in_project_and_available_to_add(self):
-        self.assertCountEqual(self.project.project_activities.all(), TaskActivityType.objects.get_defaults())
-        self.assertNotIn(self.not_default_task_activity, self.project.project_activities.all())
+    def test_manage_task_activities_view_should_add_new_task_activity_on_post_and_it_should_be_active(self):
+        task_activity_name = "New activity"
+        response = self.client.post(self.url, data={"name": task_activity_name})
```

Comment:
You should add `follow=True` parameter to this `post` method. It will allow you to check if it was really HTTP 200 status and if request was successfully done.

Reply:
Added